arelle/plugin/validate/UK/__init__.py

```diff
@@ -326,17 +327,68 @@ def checkFacts(facts):
         checkFacts(modelXbrl.facts)
 
         if val.isAccounts:
-            _missingItems = MANDATORY_ITEMS[val.txmyType] - mandatoryFacts.keys()
-            if hasCompaniesHouseContext and "UKCompaniesHouseRegisteredNumber" not in mandatoryFacts:
-                _missingItems.add("UKCompaniesHouseRegisteredNumber")
+            startDate = None
+            endDate = None
+            _missingItems = []
+            for fact in mandatoryFacts.get('EndDateForPeriodCoveredByReport'):
+                if (fact is not None and
+                        fact.xValid >= VALID and
+                        fact.context is not None and
+                        fact.context.isInstantPeriod and
+                        dateUnionEqual(fact.context.instantDate, fact.xValue)):
+                    endDate = fact.xValue
+                    break
+            if not endDate:
+                _missingItems.append('EndDateForPeriodCoveredByReport')
```

Comment:
Different validation message?  Probably confusing to see "fact missing" because the date doesn't match the fact value.